x/auth/client/rest/rest.go

```diff
@@ -4,8 +4,18 @@ import (
 	"github.com/gorilla/mux"
 
 	"github.com/cosmos/cosmos-sdk/client/context"
+	"github.com/cosmos/cosmos-sdk/x/auth/types"
 )
 
+// Codec defines the CLI codec to be used for use with the AccountRetriever.
+// Client must be sure to set this to their respective codec that implements the
+// AuthCodec interface and must be the same codec that passed to the x/auth
+// module.
+//
+// TODO:/XXX: Using a package-level global isn't ideal and we should consider
```

Comment:
can we open an issue as a proposal about this?

Reply:
I actually want to tackle it in this PR -- still thinking.